GP-138. AI: attacking

frnzee opened the pull request:
- added observe area to enemies
- added events on finding a player by enemy and for player getting out of view
- added attacking state for AI state machine
- added weapons to enemies on spawning
- small naming changes

frnzee committed:
AI attacking

- added observe area to enemies
- added events on finding a player by enemy and for player getting out of view
- added attacking state for AI state machine
- added weapons to enemies on spawning

- small naming changes

ega-ega commented on Assets/Scripts/Gameplay/AI/StateMachine/States/AttackingAIState.cs:
```diff
@@ -0,0 +1,78 @@
+            _moving = moving;
+            _observeArea = observeArea;
+
+            _observeArea.EnemyEscaped += OnEnemyEscaped;
```
Pls Add unsubscribe in launch method

ega-ega commented on Assets/Scripts/Gameplay/AI/StateMachine/States/PatrolAIState.cs:
```diff
@@ -14,23 +13,43 @@ public class PatrolAIState : IAIState
+            _observeArea = observeArea;
+            _attackingAIFactory = attackingAIFactory;
+
+            _observeArea.EnemyFound += OnEnemyFound;
```
Pls add unsubscribe

frnzee committed:
Refactoring

- combined 2 events into one
- removed unnecessary creation of moving action for stop

frnzee committed:
Refactoring

Due to unnecessarily for now:
- removed IObserveArea interface
- removed subscribing to targets changed event
- added HasTarget bool property to observe area
